userspace/libsinsp/cri.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright (C) 2013-2019 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "cri.h"
+
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+namespace libsinsp {
+namespace cri {
+std::string s_cri_unix_socket_path = "/run/containerd/containerd.sock";
+std::unique_ptr <runtime::v1alpha2::RuntimeService::Stub> s_cri = nullptr;
+int64_t s_cri_timeout = 1000;
+sinsp_container_type s_cri_runtime_type = CT_CRI;
+
+sinsp_container_type get_cri_runtime_type(const std::string &runtime_name)
+{
+	if(runtime_name == "containerd")
+	{
+		return CT_CONTAINERD;
+	} else
+	{
+		return CT_CRI;
+	}
+}
+
+bool parse_cri_image(const runtime::v1alpha2::ContainerStatus &status, sinsp_container_info *container)
+{
+	// image_ref may be one of two forms:
+	// host/image@sha256:digest
+	// sha256:digest
+	const auto &image_ref = status.image_ref();
+	auto digest_start = image_ref.find("sha256:");
+	if(digest_start != string::npos)
+	{
+		container->m_imagedigest = image_ref.substr(digest_start);
+	}
+
+	string hostname, port, digest;
+	sinsp_utils::split_container_image(status.image().image(),
+					   hostname,
+					   port,
+					   container->m_imagerepo,
+					   container->m_imagetag,
+					   digest,
```

Comment:
Should we think about cases in which the digest will be reported in `status.image().image()` but not in `status.image_ref()`? If so, in those cases `digest` should be then assigned to `container->m_imagedigest`.